### ISSUE TITLE
group messages aren't backed up during Export (likely because they're all MMS)

### ISSUE THREAD

**Issue by bobodod** [SMSSecure/SMSSecure#184]
To reproduce:

1. Create a group and send them a couple messages.
2. Navigate to Menu → Import / export → Export tab → Export plaintext backup
3. Evaluate the end of the XML file exported to the root of the SD card and gape with minimal horror at the missing messages.

It's probable that this is because MMS, even plain text ones, aren't backed up and group messages are all MMS.

**Comment by johanw666** [comment]
MMS messages are not backed up with the default backup option, wether they are group messages or not. There used to be a full backup but there were poroblems with it was removed.

**Comment by bobodod** [comment]
Pardon me for my ignorance of the underlying technology, but must group messages be MMS if they're only text?

**Comment by ArchangeGabriel** [comment]
Indeed, MMS are not exported in backup, which is a real issue. Is there anyway to backup MMS, since apparently no other apps can read SMSSecure database? (https://github.com/SMSSecure/SMSSecure/issues/192)

**Comment by ArchangeGabriel** [comment]
I suppose that backing the app folder should do it? (But only allow reimporting in SMSSecure)

**Comment by ArchangeGabriel** [comment]
Though I don’t know where that folder is, since the one in Android/ seems empty.

**Comment by johanw666** [comment]
The folder is in /data/data/org.smssecure.smssecure, but /data/data is only readable if you have root.

**Comment by ArchangeGabriel** [comment]
I think I’ve managed to back it up using `adb backup -apk -shared -all -system -f backup.ab`, without root. But I haven’t extracted the content yet, so not sure about what’s in there.

**Comment by ArchangeGabriel** [comment]
It’s empty (I mean, SMSSecure part), because SMSSecure doesn’t allow backup of its file apparently…

**Comment by BastienLQ** [comment]
@ArchangeGabriel Encrypted backups will be available in the next release, probably when #383 will be resolved.

**Comment by ArchangeGabriel** [comment]
OK, looking forward for this.

**Comment by ArchangeGabriel** [comment]
@BastienLQ Any ETA now that’s the name is changed? Just to know whether I should plan my phone reinstall this WE or next one (or even latter). ;)

**Comment by BastienLQ** [comment]
No ETA, but next release will include encrypted backups, that includes MMS messages.

**Comment by johanw666** [comment]
@ArchangelGabriel, 0.14.0 can be found on Fdroid.

**Comment by ArchangeGabriel** [comment]
OK, thanks, taking a look at it right now. We should keep this bug focused on MMS export in “SMS Backup & Restore” format.

**Comment by ArchangeGabriel** [comment]
Oh, and @bobodod: AFAIK, yes, group messages have to be MMS, SMS seems not to support multiple correspondents at all.

**Comment by johanw666** [comment]
You should do the full backup if you want mms, but that backup can not be used as an export to another app if that is what you want.

**Comment by hydrargyrum** [comment]
I want to view them on a computer, so I guess I can't decrypt the "full backup", right?

**Comment by johanw666** [comment]
Not readily, but it would be possible to write a program to do that. For Signal such a program exists but they have changed their storage format so it will definitely not work for Silence.

**Comment by hydrargyrum** [comment]
Is the format described anywhere? How to get the decryption key?

**Comment by johanw666** [comment]
The format is documented by the source code. I believe you need to get masterkey to decrypt the stored messages.

**Comment by hydrargyrum** [comment]
> The format is documented by the source code.

... that's not "documentation".

>  I believe you need to get masterkey to decrypt the stored messages.

How?

**Comment by hydrargyrum** [comment]
So, any news on this issue?
